vg/draw/draw_test.go

```diff
@@ -115,3 +116,44 @@ func TestTile(t *testing.T) {
 		}
 	}
 }
+
+func TestFormattedCanvas(t *testing.T) {
+	for _, test := range []struct {
+		format string
+		err    error
+	}{
+		{format: "eps"},
+		{format: "jpg"},
+		{format: "jpeg"},
+		{format: "pdf"},
+		{format: "png"},
+		{format: "svg"},
+		{format: "tex"},
+		{format: "tiff"},
+		{format: "tif"},
+		{
+			format: "",
+			err:    fmt.Errorf("unsupported format: \"\""),
+		},
+		{
+			format: "abc",
+			err:    fmt.Errorf("unsupported format: \"abc\""),
+		},
+	} {
+		t.Run(test.format, func(t *testing.T) {
+			_, err := NewFormattedCanvas(10, 10, test.format)
+			switch {
+			case err != nil && test.err != nil:
+				if got, want := err.Error(), test.err.Error(); got != want {
+					t.Fatalf("invalid error.\ngot= %v\nwant=%v", got, want)
+				}
+			case err != nil && test.err == nil:
+				t.Fatalf("unexpected error: %+v", err)
+			case err == nil && test.err != nil:
+				t.Fatalf("expected an error (got=%v)", err)
+			case err == nil && test.err == nil:
+				// ok
```

Comment:
The set of cases is exhaustive, so this can be omitted since there's no default to properly condition, no?

Reply:
sure.
I just wanted the `switch` to be thorough (and explicit).
but ok.